TrenDAP/wwwroot/TypeScript/Features/Widgets/Implementations/Map.tsx

```diff
@@ -294,34 +339,37 @@ export const Map: WidgetTypes.IWidget<IProps, IChannelSettings, null> = {
                 valueField: 'Frequency'
             };
 
-            const heatmapLayer = new HeatMap(cfg);
-            heatmapLayer.setData({
+            heatmapLayer.current = new HeatMap(cfg);
+            heatmapLayer.current.setData({
                 data: props.Data
                     .filter(d => d.Latitude != null && d.Longitude != null)
                     .map(d => ({ Name: d.Name, Frequency: d.SeriesData[d.ChannelSettings.Field]?.[0]?.[1], Lat: d.Latitude, Long: d.Longitude }))
             });
-            heatmapLayer.addTo(map.current);
+            heatmapLayer.current.addTo(map.current);
 
-            return () => { map.current.removeLayer(heatmapLayer); }
+            return () => { map.current.removeLayer(heatmapLayer.current); }
         }, [props.Data, props.Settings.Type]);
 
         React.useEffect(() => {
             // Clear and reset animation
             clearInterval(animationTimeout.current.timeout);
             animationTimeout.current.animationTime = 0;
 
-            if (map.current == null || props.Settings.Type === 'HeatMap') return;
+            if (map.current == null) return;
 
             // Only need to set the interval if we need to animate
             if (isAnimated) {
                 animationTimeout.current.timeout = setInterval(() => {
                     animationTimeout.current.animationTime = (animationTimeout.current.animationTime + animationStep / 1000) % props.Settings.AnimationTime;
                     const animationTimeStamp = (animationTimeout.current.animationTime / props.Settings.AnimationTime) * (limitValues.end - limitValues.start) + limitValues.start;
-                    updateCircles(animationTimeStamp, false);
+                    if (props.Settings.Type === 'HeatMap')
+                        updateHeatmap(animationTimeStamp,false);
+                    else
+                        updateCircles(animationTimeStamp, false);
                 }, animationStep);
             }
             return updateCircles(0, true);
-        }, [updateCircles, props.Settings.Type, isAnimated, props.Settings.AnimationTime]);
+        }, [updateCircles, updateHeatmap, props.Settings.Type, isAnimated, props.Settings.AnimationTime]);
```

Comment:
UI needs a small update with this, it looks like it still locks out the animation settings if we are in heat map.